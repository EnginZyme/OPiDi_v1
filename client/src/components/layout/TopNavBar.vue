<template>
  <div>
    <div class="top_nav_bar">
      <div class="top_nav_bar__content">
        <div></div>
        <AppBanner :showImage="false" />
      </div>
    </div>
  </div>
</template>

<script>
import AppBanner from "../base/AppBanner";

/**
 * A component that renders the top navigation bar.
 */
export default {
  name: "TopNavBar",
  components: {
    AppBanner,
  },
};
</script>

<style scoped>
.top_nav_bar {
  height: 50px;
  background: rgb(23, 26, 46);
  background: linear-gradient(
    180deg,
    rgba(23, 26, 46, 0.9603991254705007) 0%,
    rgba(23, 26, 46, 1) 74%
  );
  filter: drop-shadow(0px 4px 4px rgba(0, 0, 0, 0.25));
  display: grid;
}
.top_nav_bar__content {
  display: flex;
  flex-direction: row;
  align-content: center;
  justify-content: space-between;
  margin-left: 28px;
  margin-right: 28px;
}
</style>
