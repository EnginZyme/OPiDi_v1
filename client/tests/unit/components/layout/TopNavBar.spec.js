import { shallowMount } from "@vue/test-utils"
import TopNavBar from "@/components/layout/TopNavBar"


describe("shallowMounted TopNavBar", () => {
    const wrapper = shallowMount(TopNavBar)

    it("renders correctly", () => {
        expect(wrapper.element).toMatchSnapshot();
    })
})
