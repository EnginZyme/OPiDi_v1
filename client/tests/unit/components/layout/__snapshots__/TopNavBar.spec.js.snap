// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallowMounted TopNavBar renders correctly 1`] = `
<div>
  <div
    class="top_nav_bar"
  >
    <div
      class="top_nav_bar__content"
    >
      <div />
       
      <appbanner-stub />
    </div>
  </div>
</div>
`;
